org.eclipse.xtend.m2e/src/org/eclipse/xtend/m2e/XtendProjectConfigurator.java

```diff
@@ -158,20 +158,19 @@ private <T> T mojoParameterValue(String paramName, Class<T> paramType, ProjectCo
 	}
 
 	static IProject getProject(ProjectConfigurationRequest request) {
-		// DO NOT USE A METHOD REFERENCE!
-		return call(request, r -> r.getMavenProjectFacade(), "mavenProjectFacade").getProject();
+		return XtendProjectConfigurator.<ProjectConfigurationRequest, IMavenProjectFacade>call(request, "getMavenProjectFacade", "mavenProjectFacade").getProject();
 	}
 
 	static MavenProject getMavenProject(ProjectConfigurationRequest request) {
-		// DO NOT USE A METHOD REFERENCE!
-		return call(request, r -> r.getMavenProject(), "mavenProject");
+		return call(request, "getMavenProject", "mavenProject");
 	}
 
 	@SuppressWarnings("unchecked")
-	private static <T, R> R call(T obj, Function<T, R> getter, String newMethodName) {
+	private static <T, R> R call(T obj, String oldMethodName, String newMethodName) {
```

Comment:
Alternatively we could use a varargs `methodNames` String arguments:
```suggestion
	private static <T, R> R call(T obj, String... methodNames) {
```
and then just loop over them and try each and throw an exception if none worked:
```
for (String methodName: methodNames) {
	try {
		Method method = obj.getClass().getMethod(oldMethodName);
		return (R) method.invoke(obj);
	} catch (Error e) { // try next one
	}
}
throw new IllegalStateException("Unable to call " + methodNames[0] + "on " + obj ,e);
```
Just coded this in the browser so this might need some formatting.